frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/toolbar/ResetCurrentBranchToRemoteAction.java

```diff
@@ -0,0 +1,30 @@
+package com.virtuslab.gitmachete.frontend.actions.toolbar;
+
+import static com.virtuslab.gitmachete.frontend.actions.common.ActionUtils.getCurrentBranchNameIfManaged;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import io.vavr.control.Option;
+
+import com.virtuslab.gitmachete.frontend.actions.common.BaseResetBranchToRemoteAction;
+import com.virtuslab.gitmachete.frontend.datakeys.DataKeys;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_GIT_MACHETE_REPOSITORY}</li>
+ *  <li>{@link DataKeys#KEY_SELECTED_VCS_REPOSITORY}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public class ResetCurrentBranchToRemoteAction extends BaseResetBranchToRemoteAction {
+  @Override
+  public Option<String> getNameOfBranchUnderAction(AnActionEvent anActionEvent) {
+    return getCurrentBranchNameIfManaged(anActionEvent);
+  }
+
+  @Override
+  protected String getActionType() {
```

Comment:
not needed as we can retrieve action place from `anActionEvent` (see other `IBranchNameProvider` implementations)

Reply:
So it seems that comments are ordered by datetime.... :thinking:  cool !

Reply:
Ok, I give up this distinction completely.
Now I just show branch name instead of it's type (current/selected)

Reply:
You can leave the distinction but just base it on `com.intellij.openapi.actionSystem.AnActionEvent#getPlace` rather than a bespoke method of yours